app/models/report.rb

```diff
@@ -4,24 +4,44 @@ class Report
   attr_reader :exercise_logs, :pain_logs, :pt_sessions, :pains, :exercises, :body_parts
 
   def initialize(args)
-    @filter_params = args
-    @exercise_logs ||= query_logs('exercise_logs')
-    @pain_logs ||= query_logs('pain_logs')
-    @pt_sessions ||= query_logs('pt_sessions')
-    @pains ||= @filter_params[:user].pains
-    @exercises ||= @filter_params[:user].exercises
-    @body_parts ||= @filter_params[:user].body_parts
+    @exercise_logs = args[:exercise_logs]
+    @pain_logs = args[:pain_logs]
+    @pt_sessions = args[:pt_sessions]
+    @pains = args[:pains]
+    @exercises = args[:exercises]
+    @body_parts = args[:body_parts]
   end
```

Comment:
Rubocop said this was too complex. I agreed and broke it out into a separate method that didn't memoize. I moved the memoization into the controller.

Reply:
I like where this is headed! It looks like `Report` gets initialized with the `filter_params` and so it makes sense to pass those in and set an instance variable for them in the initializer.

```
def initializer(filter_params)
  @filter_params = filter_params
end
```

And then it looks like we're setting all of the rest of the state for this object in the `self.build_report` method, but what if we moved all of those instance vars into individual methods and memoized them there since there's some query action happening.

```
def exercise_logs
  @exercise_logs ||= query_logs('exercise_logs')
end

def pains
  @pains ||= filter_params[:user].pains
end
```

Then we could make the `query_logs` method a private method on `Report` and I think you'd end up with the functionality you're looking for!

Reply:
Yesssss. That is what I want. :heart_eyes_cat: